xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h

```diff
@@ -343,6 +343,15 @@ extern "C"
    */
   PVR_ERROR GetRecordingEdl(const PVR_RECORDING& recording, PVR_EDL_ENTRY edl[], int *size);
 
+  /*!
+   * Retrieve the size of a recording on the backend.
+   * @param recording The recording.
+   * @return The size in bytes or -1 on error
+   * @remarks Required if bSupportsRecordingSize is set to true.
+   *          Return -1 if this add-on won't provide this function.
+   */
+  int64_t GetRecordingSize(const PVR_RECORDING& recording);
```

Comment:
1. All API functions should return PVR_ERROR and return values as out parameters (pointer types in C, no refs please as they are C++ only). With this, it is possible for the addon the report for instance lost connection to backend etc.

2. Pass the recoring as pointer, not reference. Refs are C++, this is a C API.

BTW: I know we do not always follow this rules, but we should make it better when introducing new API functions.

Reply:
Ok, I understand. So eventually when everything moves to the C++ API the underlying C API can be fixed.